pennylane_lightning_kokkos/src/tests/Test_StateVectorKokkos_Expval.cpp

```diff
@@ -200,3 +200,87 @@ TEMPLATE_TEST_CASE("StateVectorKokkosManaged::getExpectationValueTwoQubitOp",
+        CHECK(real(expected) == Approx(results).epsilon(1e-7));
+    }
+
+    SECTION("GetExpectionHermitianMatrix") {
+        using cp_t = Kokkos::complex<TestType>;
+        std::vector<cp_t> init_state{{0.0, 0.0}, {0.0, 0.1}, {0.1, 0.1},
+                                     {0.1, 0.2}, {0.2, 0.2}, {0.3, 0.3},
+                                     {0.3, 0.4}, {0.4, 0.5}};
+        StateVectorKokkos<TestType> kokkos_sv{init_state.data(),
+                                              init_state.size()};
+        std::vector<size_t> wires{0, 1, 2};
+        std::vector<cp_t> matrix{
+            {0.5, 0.0},  {0.2, 0.5},  {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5},
+            {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},
+            {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5},
+            {0.3, 0.0},  {0.5, 0.0},  {0.2, 0.5},  {0.2, -0.5}, {0.3, 0.0},
+            {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5},
+            {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},
+            {0.2, -0.5}, {0.3, 0.0},  {0.5, 0.0},  {0.2, 0.5},  {0.2, -0.5},
+            {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},
+            {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5},
+            {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},  {0.5, 0.0},  {0.2, 0.5},
+            {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5},
+            {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0},
+            {0.2, -0.5}, {0.3, 0.0},  {0.2, -0.5}, {0.3, 0.0}};
+
+        auto results = kokkos_sv.getExpectationValue(wires, matrix);
+        cp_t expected(1.263000, -1.011000);
+
+        CHECK(real(expected) == Approx(results).epsilon(1e-7));
+    }
+}
+
+TEMPLATE_TEST_CASE("StateVectorKokkos::Hamiltonian_expval_Sparse",
+                   "[StateVectorKokkos_Nonparam]", float, double) {
+    using cp_t = Kokkos::complex<TestType>;
+    SECTION("GetExpectionSparse") {
+        std::vector<cp_t> init_state{{0.0, 0.0}, {0.0, 0.1}, {0.1, 0.1},
+                                     {0.1, 0.2}, {0.2, 0.2}, {0.3, 0.3},
+                                     {0.3, 0.4}, {0.4, 0.5}};
+        StateVectorKokkos<TestType> kokkos_sv{init_state.data(),
+                                              init_state.size()};
+
+        std::vector<size_t> index_ptr = {0, 2, 4, 6, 8, 10, 12, 14, 16};
+        std::vector<size_t> indices = {0, 3, 1, 2, 1, 2, 0, 3,
+                                       4, 7, 5, 6, 5, 6, 4, 7};
+        std::vector<Kokkos::complex<TestType>> values = {
+            {1.0, 0.0},  {0.0, -1.0}, {1.0, 0.0}, {0.0, 1.0},
+            {0.0, -1.0}, {1.0, 0.0},  {0.0, 1.0}, {1.0, 0.0},
+            {1.0, 0.0},  {0.0, -1.0}, {1.0, 0.0}, {0.0, 1.0},
+            {0.0, -1.0}, {1.0, 0.0},  {0.0, 1.0}, {1.0, 0.0}};
+
+        auto result = kokkos_sv.getExpectationValue(values, indices, index_ptr);
+
+        TestType expected = 1;
```

Comment:
I'd also suggest an expval test that doesn't rely on 0 or 1 as the result value. Some arbitrary real may be safer, and may help avoid similar issues we hit in the the LightningGPU implementation